homeassistant/components/fritz/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""AVM FRITZ!Box connectivitiy sensor."""
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_CONNECTIVITY,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .common import FritzBoxHostEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug("Setting up FRITZ!Box binary sensors")
+    fritzbox_tools = hass.data[DOMAIN][entry.entry_id]
+
+    if "WANIPConn1" in fritzbox_tools.connection.services:
+        # Only routers are supported at the moment
+        async_add_entities(
+            [FritzBoxConnectivitySensor(fritzbox_tools, entry.title)], True
+        )
+
+
+class FritzBoxConnectivitySensor(FritzBoxHostEntity, BinarySensorEntity):
```

Comment:
We should have a separate base class for entities that are on the fritz box itself vs entities that are other devices

Reply:
Done